### ISSUE TITLE
navi export assets  and include last credentialed scan?

### ISSUE THREAD

**Issue by armada-lmt** [packetchaos/navi#42]
Is it possible to get "last credentialed scan" from navi export assets?  I see last licensed scan... but not all licensed scans are credentialed.

**Comment by packetchaos** [comment]
Yes it is possible.  It requires querying the vulns table and joining it with the assets table on the asset_uuid.  If you send me what data you want I can provide an example query.  There are a few methods.  One would be checking for a plugin which confirms a credential was used and the second is looking in the 19506 output for "credentialed=yes".  

There may be other methods.

**Comment by armada-lmt** [comment]
Sorry just saw your reply now while researching some issues.  So Here's why I wanted last credentialed scan.  We use this tool to monitor and ensure servers are getting regular vuln scans.  We are using "navi export assets" which includes "last licensed scan".  Unfortunately, I found a host where the last licensed scan is current (9/3), BUT when you look at it in the UI, the authentication for the scan failed... so effectively, it was not scanned (the OS isn't even identified correctly).  
![image](https://github.com/user-attachments/assets/4c64cbe6-c866-4be9-bdfe-addb6e4982d0)

I don't see "authentication status" exported from navi.  So not only would I need "last authenticated scan" timestamp exported, I would need the status exported to determine if that scan was successful or not.

At this time, as it is, I can't trust any of the last scan output from navi export assets at all... unless I'm missing something?

thanks!

**Comment by armada-lmt** [comment]
Any feedback on the previous question?

**Comment by packetchaos** [comment]
There are a couple of way's to accomplish this task.  I think the easiest I could think of was to use a custom query looking for any asset that does not have 104410 present.  This plugin fires when the credentials have failed.  However, keying on this value alone will lump in discovery scans.  So you have to have two queries joined together.

Essentially you need to look at assets where "Credentialed Checks : yes" in the 19506 plugin and the absence of 104410.  To test your query before export

`navi find query "select * from vulns where (plugin_id='19506' and output LIKE '%Credentialed checks : yes%') and plugin_id !='104410';"`

To Export it:

`navi export query "select * from vulns where (plugin_id='19506' and output LIKE '%Credentialed checks : yes%') and plugin_id !='104410';"`

This would give you all of the plugins on all of the assets in the scan.  If you wanted just the assets you would have to join the tables on asset_uuid(vulns table) and uuid(assets table).
